xbmc/utils/StringUtils.cpp

```diff
@@ -1694,7 +1694,12 @@ void StringUtils::WordToDigits(std::string &word)
 std::string StringUtils::CreateUUID()
 {
 #ifdef HAVE_NEW_CROSSGUID
+#ifdef TARGET_ANDROID
+  JNIEnv* env = xbmc_jnienv();
```

Comment:
Looking at the implementation of xg::newGuid, is the global init call xg::initJni in CXNMCApp ctor still needed?

Reply:
For whatever reason, yes it is already. As far as i could tell theres  actually no reason for it currently to be there. Didnt bother looking through history of the file
https://github.com/xbmc/xbmc/blob/58884ca25a5567286f05bc826cffc5640fbfc5b0/xbmc/utils/StringUtils.cpp#L25-L27

According to crossguid docs, yes it is still required

```
NOTE: On Android you need to call xg::initJni(JNIEnv *) first so that it is possible for xg::newGuid()
to call back into java libraries. initJni only needs to be called once when the process starts.
```

My understanding after looking at the PR that introduced this behaviour https://github.com/graeme-hill/crossguid/pull/24
The overload newGuid(JNIEnv *env) was introduced to handle multi threaded use of the lib on android. Honestly, i dont want to dig any further into why things worked the way they did, or how. The issue that was closed as part of that PR was https://github.com/graeme-hill/crossguid/issues/23 It has reasoning from the crossguid dev and the PR author. Feel free to bring up an issue with crossguid upstream if you feel strongly about it.